src/tools/wasm-interp.cc

```diff
@@ -161,33 +175,126 @@ Result RunAllExports(const Instance::Ptr& instance, Errors* errors) {
+          Trap::Ptr* trap) -> Result {
+        printf("called host ");
+        WriteCall(stream, import_name, func_type, params, results, *trap);
+        return Result::Ok;
+      });
+
+  return host_func.ref();
+}
+
+static Extern::Ptr GetImport(const std::string& module,
+                             const std::string& name) {
+  auto mod_iter = s_registry.find(module);
+  if (mod_iter != s_registry.end()) {
+    auto extern_iter = mod_iter->second.find(name);
+    if (extern_iter != mod_iter->second.end()) {
+      return extern_iter->second;
+    }
+  }
+  return {};
+}
+
+static void PopulateExports(const Instance::Ptr& instance,
+                            Module::Ptr& module,
+                            ExportMap& map) {
+  map.clear();
+  // Module::Ptr module{s_store, instance->module()};
+  for (size_t i = 0; i < module->export_types().size(); ++i) {
+    const ExportType& export_type = module->export_types()[i];
+    auto export_ = s_store.UnsafeGet<Extern>(instance->exports()[i]);
+    map[export_type.name] = export_;
+  }
+}
+
+static std::string GetPathName(std::string module_arg) {
+  size_t path_at = module_arg.find_first_of(':');
+
+  path_at = path_at == std::string::npos ? 0 : path_at + 1;
+
+  return module_arg.substr(path_at);
+}
+
+static std::string FileNameOfPath(std::string path_name) {
+  // use file_name (without extension)
+  size_t fstart = path_name.find_last_of("/\\");
+  size_t fend = path_name.find_last_of(".");
+
+  fstart = fstart == std::string::npos ? 0 : fstart;
+  fend = fend < fstart ? std::string::npos : fend;
+
+  return path_name.substr(fstart, fend);
+}
+
+static std::string GetRegistryName(std::string module_arg,
+                                   const Module::Ptr& module) {
+  size_t split_pos = module_arg.find_first_of(':');
+  if (split_pos == std::string::npos) {
+    split_pos = 0;
+  }
+  std::string override_name = module_arg.substr(0, split_pos ? split_pos : 0);
+  std::string path_name = module_arg.substr(split_pos);
+  std::string debug_name = "";  // GetDebugName(module);
+
+  // check if override_name starts with @ and return override_name
+  if (override_name[0] == '@') {
+    if (override_name.size() > 1) {
+      return override_name.substr(1);
+    }
+    // ignore debug-name and use filename
+    return FileNameOfPath(path_name);
+  }
+
+  // if no override_name is provided -> use debug name or filename
+  if (override_name.empty()) {
+    // prefer debug_name, if no override_name is provided
+    if (!debug_name.empty()) {
+      return debug_name;
+    }
+    // use file_name (without extension)
+    return FileNameOfPath(path_name);
+  }
+
+  // prefer debug_name if override_name starts with '$'
+  if (override_name[0] == '$' && !debug_name.empty()) {
+    return debug_name;
+  }
+
+  // return the override_name (remove leading '$' if present)
+  bool leading_dollar = override_name[0] == '$';
+  return override_name.substr(leading_dollar ? 1 : 0);
```

Comment:
Do we need this dollar sign syntax?    What about this simpler scheme:

1. Override name, if present
2. Debug name, if present
3. Fall back to file name

Reply:
We need something to explicitly go for the debug name. I dont want something unexpected. Like downloading a module from the web, and not have it use the filename, because you might not check for the debug name. (just for convenience)

I came up with
1. `override_name` if present
2. if `override_name == "$"` debug_name if present
3. Fall back to file_name

```cpp
  if (!override_name.empty()) {
    // prefer override_name unless its just "$"
    if (override_name != "$") {
      return override_name;
    }
    
    // prefer debug_name if override_name == "$"
    if (!debug_name.empty()) {
      return debug_name;
    }
  }
  
  // fall back to file-name
  return FileNameOfPath(path_name);
```

Reply:
or we go for this:
1. if `override_name == "@"` use `file_name`
2. `overriide_name` if present
3. `debug_name` if present
4. Fall back to `file_name`

Binary compatibility woutl then be prefered

Reply:
I just don't think we need a way to force the file name to be used over the debug name.   If you really want to first the filename why not just use the override mechism.  e.g -m libfoo:libfoo.wasm.

In other words, always prefer the debug name, but give users an opt out to override if they want to.  That would avoid the magic `@` thing.

Reply:
was just for conveniance... But tbh I don't care to much.
Its done already. I removed the magic thingy.

Reply:
Oh I also can't find a way to get the DebugName, my best guess was it's hidden somewhere in `Instance`, but I cant find it.

Reply:
@sbc100 minor question, wouldn't it make sense to just ... use all three? Like: Always register by `file_name`, if exists register by `debug_name`, if exists register by `override_name`.
This could be an idea for a revision. And as soon as I findout how to get `debug_name`. (It seems like I need to query it manually from the file? The other approaches used in other files are ... complicated)